app/models/life_cycle.rb

```diff
@@ -0,0 +1,47 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+class LifeCycle < ApplicationRecord
+  has_many :project_life_cycles, class_name: "Project::LifeCycle", dependent: :destroy
+  has_many :projects, through: :project_life_cycles
+
+  belongs_to :color, optional: false
+
+  validates :name, presence: true
+  validates :type, inclusion: { in: %w[Stage Gate], message: :must_be_a_stage_or_gate }
```

Comment:
I agree, the reason behind adding this validation is that the `type` will be a user supplied value and in case it is incorrect, the page will fail with an 500 error. Having this validation will gracefully handle the eventual type mismatch coming from the UI and will display the error message.
Perhaps an improvement would be to add the error to the `:base` and not on the `:type`, as we won't display the type as a separate input field on the LifeCycle creation form.